### ISSUE TITLE
Bug: `key` of constructor in `class C { "constructor"() {} }` should not be an `Identifier`

### ISSUE THREAD

**Issue by overlookmotel** [typescript-eslint/typescript-eslint#11084]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&showAST=es&fileType=.tsx&code=MYGwhgzhAEDC0G8BQ1oCJgHsB2EAuATgK7B6YFoAUAlItAL5KNA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class C {
  "constructor"() { }
}
```

### ESLint Config

```javascript
n/a
```

### tsconfig

```jsonc
n/a
```

### Expected Result

I expected `key` property of `MethodDefinition` to be a `Literal`, not an `Identifier`.

This would match Espree ([AST explorer](https://ast-explorer.dev/#eNo9jTEKwzAMRa9iNLWQE2TtFUonL0bVkKJYRpILxeTuVUgIaND/7yENYJjhU77FUJfmMEGLgqwpUQSMgFzM0iONXFPKgFLNtaOLZrjd09hyjQlZQj4lk65Iz1+jDHPkVd6dY58OTLiWF6ktUg/Oxcn84iy49/GG4jhsf6HRNjo=)).

### Actual Result

`key` property of `MethodDefinition` is an `Identifier`:

```json
{
  "type": "MethodDefinition",
  "key": {
    "type": "Identifier",
    "range": [12, 25],
    "name": "constructor",
    "decorators": [],
    "optional": false,
    "typeAnnotation": null
  },
  "kind": "constructor",
  "...": "..."
}
```

Main reason I think this is a bug is that it's different from Espree, where `key` is a `Literal`.

But additionally, I feel this is error-prone, because you can normally assume that the section of source text corresponding to an `Identifier`'s range starts with either a *UnicodeIDStart* character, or `\`. This breaks that rule - the first character is `"`.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/parser`            | `8.30.1` |

**Comment by overlookmotel** [comment]
Great! Thanks for fixing @ntdiary.

Can I ask: @bradzacher Do you have a guess as to when next release will be?

**Comment by JoshuaKGoldberg** [comment]
In a couple of hours. 😁 

https://typescript-eslint.io/users/releases#latest

**Comment by overlookmotel** [comment]
Ha! It wasn't so hard to guess! Automatically once a week is a nice release policy.
